tempodb/encoding/common/interfaces.go

```diff
@@ -43,6 +43,8 @@ type SearchOptions struct {
 	ReadBufferCount        int
 	ReadBufferSize         int
 	BlockReplicationFactor int // Only blocks with this replication factor will be searched. Set to 1 to search generator blocks (RF=1).
+	Limit                  uint32
```

Comment:
i'd prefer not adding these here since these fields only impact tag values and not normal search. it looks like they were added b/c of this code:

https://github.com/grafana/tempo/pull/4320/files#diff-d68ed2a23a87b802e3d85870a36046be8d2f61dd2caeab632503164c9b0ab57dR384

Can we construct the distinct collector thing outside the call and pass it in? is there a clean way to avoid adding these fields?

Reply:
I have opted for a different solution, I'm passing the request in the same way we are doing for `SearchTagValuesV2` that way we standardize the interface contract

Reply:
very clean fix. love it ❤️